examples/SmartShop/contracts/SellerContract.aes

```diff
@@ -1,35 +1,36 @@
 payable contract Seller =
   record state = 
     { seller_address : address,
+      buyer_address : address,
       status : string,
-      contract_address : address }
+      price : int }
   
-  stateful entrypoint init() : state =
+  entrypoint init(ba : address, price' : int) : state =
     { seller_address = Call.caller,
+      buyer_address = ba,
       status = "undefined",
-      contract_address = Contract.address }
+      price = price' }
 
   function check_money() : bool =
-    Contract.balance > 0
+    Contract.balance >= state.price
     
   stateful entrypoint send_item() : string =
     require(check_money(), "Not enough money")
     put(state{status = "sent_to_transport_courier"})
     state.status
+
+  function check_buyer() : bool =
```

Comment:
`check_buyer()` is not a very self-explanatory naming for the purpose of this function.

Reply:
How about check_state_buyer().

Also for this function in question, the best would have been to check if the `Call.caller == state.buyer_address` but for a reason or the other, this is not executing even after deploying the BuyerContract with another wallet keyPair at the unit test script.

Reply:
Fixed in the new commit